lib/features/keyboard/Keyboard.js

```diff
@@ -91,6 +104,43 @@ Keyboard.$inject = [
   'editorActions'
 ];
 
+Keyboard.prototype._keyHandler = function(event) {
+
+  var target = event.target,
+      key = getKeyValue(event),
+      eventBusResult,
+      globalListenersResult;
+
+  if (target && (domMatches(target, 'input, textarea') || target.contentEditable === 'true')) {
+    return;
+  }
+
+  var eventType = getKeydownEventType(key);
+
+  var context = {
+    event: event,
+    key: key
+  };
+
+  eventBusResult = this._eventBus.fire(eventType, context);
+
+  // run global listeners only if no listeners for specific keys reacted
+  if (!eventBusResult) {
+    forEach(this._globalListeners, function(listener) {
```

Comment:
Global listeners are being called with the code while local listeners are being called with the key? I'd rethink this behavior as I don't think it is intuitive.

Reply:
I agree that it's not intuitive. We use `keyCode` for `globalListeners` only in order to preserve the backward compatibility, i.e. to not break for example the [BpmnKeyBindings](https://github.com/bpmn-io/bpmn-js/blob/master/lib/features/keyboard/BpmnKeyBindings.js).
Alternatively, we could drop the old keyboard related behaviour as `KeyboardEvent.keyCode` is already a `deprecated` feature.

Reply:
There is no blog post regarding the way to add a keyboard listener ([one mention of the feature](https://bpmn.io/blog/posts/2016-copy-paste-search-bpmn-js-0-15.html)); the examples don't address this issue either. 
There are two forum topics regarding the keyboard issues: [1](https://forum.bpmn.io/t/copy-paste-and-keyboard-events/741/3), [2](https://forum.bpmn.io/t/copy-and-paste-keyboard-binding-and-eventbus-on-keyboard-event-register/2005)

The change to `key` only would break `BpmnKeyBindings` as it uses `keyCode`. However, the feature itself needs refactor as it is a one big listener for each of Bpmn-js-specific key bindings.